### PR TITLE
Refine: error handle in cita_executive.

### PR DESCRIPTION
**Pull request by leeyr338**
- Refine: error handle in cita_executive.

### CODE REVIEW

File: cita-executor/core/src/cita_executive.rs [hunk 1]
```diff
@@ -101,7 +101,7 @@ impl<'a, B: DB + 'static> CitaExecutive<'a, B> {
             //        got: U256,
             //    }
             // Need to change VMError defined in cita-vm.
-            return Err(ExecutionError::VM(VMError::NotEnoughBaseGas));
+            return Err(ExecutionError::NotEnoughBaseGas);
```
Comment (ouwenkg) [inline cita-executor/core/src/cita_executive.rs]: remove the comment in last line.

File: cita-executor/core/src/cita_executive.rs [hunk 2]
```diff
@@ -786,6 +793,41 @@ impl Default for EnvInfo {
...
+        ExecutedException::VM(err)
+    }
+}
+
+impl From<NativeError> for ExecutedException {
+    fn from(err: NativeError) -> Self {
+        ExecutedException::NativeContract(err)
+    }
+}
+
```
Comment (ouwenkg) [inline cita-executor/core/src/cita_executive.rs]: Put all error handling logic in another seperated file is better.

File: cita-executor/core/src/cita_executive.rs [hunk 3]
```diff
@@ -639,56 +651,51 @@ pub fn contract_address(address: &Address, nonce: &U256) -> Address {
...
+    /// Return when authentication error occurs.
     Authentication(AuthenticationError),
-    NativeContract(NativeError),
+    /// Return when the quota_limit in transaction lower then base quota required.
+    NotEnoughBaseGas,
+    /// Return when transaction nonce does not match state nonce.
+    InvalidNonce,
+    /// Return when the cost of transaction (value + quota_price * quota) exceeds.
+    NotEnoughBalance,
+    /// Return when the block quota exceeds block quota limit.
```
Comment (ouwenkg) [inline cita-executor/core/src/cita_executive.rs]: I am a little confused about what the difference between `ExecutionError` and `ExecutedException`.

`ExecutedException` wraps error returned from vm?  if this logic is right,  Maybe `InvalidNonce` and `NotEnoughBalance` belongs to `ExecutedException`.

Comment (leeyr338) [reply cita-executor/core/src/cita_executive.rs]: `ExecutedException` will be set into `Receipt`, but `ExecutionError` will be handled in `block.rs` where called the `exec` function.

File: cita-executor/core/src/cita_executive.rs (hunk 2 above)
Comment (leeyr338) [reply cita-executor/core/src/cita_executive.rs]: This is some errors about `executive`, I will think about it later!

File: cita-executor/core/src/cita_executive.rs (hunk 1 above)
Comment (leeyr338) [reply cita-executor/core/src/cita_executive.rs]: Why? This is a FIXME comment, and I also think the VMError should be redefined.

**Pull request merged**